pkg/api/repositories.go

```diff
@@ -36,56 +36,112 @@ type RepositoryResponse struct {
-	URL                  *string   `json:"url"`                                               // URL of the remote yum repository
-	DistributionVersions *[]string `json:"distribution_versions" example:"7,8"`               // Versions to restrict client usage to
-	DistributionArch     *string   `json:"distribution_arch" example:"x86_64"`                // Architecture to restrict client usage to
-	GpgKey               *string   `json:"gpg_key"`                                           // GPG key for repository
-	MetadataVerification *bool     `json:"metadata_verification"`                             // Verify packages
-	ModuleHotfixes       *bool     `json:"module_hotfixes"`                                   // Disable modularity filtering on this repository
-	Snapshot             *bool     `json:"snapshot"`                                          // Enable snapshotting and hosting of this repository
-	AccountID            *string   `json:"account_id" readonly:"true" swaggerignore:"true"`   // Account ID of the owner
-	OrgID                *string   `json:"org_id" readonly:"true" swaggerignore:"true"`       // Organization ID of the owner
-	Origin               *string   `json:"origin" readonly:"true" swaggerignore:"true"`       // Origin of the repository
-	ContentType          *string   `json:"content_type" readonly:"true" swaggerignore:"true"` // Content Type (rpm) of the repository
+	UUID        *string `json:"uuid" readonly:"true" swaggerignore:"true"`
+	AccountID   *string `json:"account_id" readonly:"true" swaggerignore:"true"`   // Account ID of the owner
+	OrgID       *string `json:"org_id" readonly:"true" swaggerignore:"true"`       // Organization ID of the owner
+	Origin      *string `json:"origin" readonly:"true"`                            // Origin of the repository
+	ContentType *string `json:"content_type" readonly:"true" swaggerignore:"true"` // Content Type (rpm) of the repository
+
+	Name                 *string   `json:"name"`                                // Name of the remote yum repository
+	URL                  *string   `json:"url"`                                 // URL of the remote yum repository
+	DistributionVersions *[]string `json:"distribution_versions" example:"7,8"` // Versions to restrict client usage to
+	DistributionArch     *string   `json:"distribution_arch" example:"x86_64"`  // Architecture to restrict client usage to
+	GpgKey               *string   `json:"gpg_key"`                             // GPG key for repository
+	MetadataVerification *bool     `json:"metadata_verification"`               // Verify packages
+	ModuleHotfixes       *bool     `json:"module_hotfixes"`                     // Disable modularity filtering on this repository
+	Snapshot             *bool     `json:"snapshot"`                            // Enable snapshotting and hosting of this repository
+}
+
+type RepositoryUpdateRequest struct {
```

Comment:
Not saying I'm opposed, just curious, why separate the update request struct? I guess one benefit might be making the API more readable.

Reply:
While making the origin parameter available for create, i didn't want it available for update.  There's no way in openapi to indicate that a parameter is only for creation, and i saw some discussion mentioning that using different structs was the correct way to handle it.   We also did it this way for template.

That said, it complicates and duplicates the code quite a bit.  I tried using inheritance and that reduced the duplication a bit, but increased complexity and i didn't think was an improvement.